transformer_engine/common/common.h

```diff
@@ -599,4 +599,14 @@ bool is_supported_by_CC_100();
 
 }  // namespace transformer_engine
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+bool nvte_is_non_tn_fp8_gemm_supported();
+
+#ifdef __cplusplus
+}  // extern "C"
+#endif
```

Comment:
Exposing this as a C function is strange to me. We do that for the publicly exposed functions so that we support C users, but this is a utility function that we only expect to use within C++.

I think the right approach is to put this within the `transformer_engine` namespace. I am thinking about adding some cuBLAS utility functions to help with https://github.com/NVIDIA/TransformerEngine/issues/1585, so I may move this there in the future.

Reply:
If we add it to the `transformer_engine` namespace, we'll end up exposing this function in `lib_transformerengine.version` as 
```
              transformer_engine::nvte_cudnn_handle_init*;
```

and I believe we don't want to do that

Reply:
Although I do agree that putting a single C function there does seem odd - as you suggested maybe a utility functions file would be better but even then it'd be a separate file with a single utility function

Reply:
We're already exposing the function, so we might as well make it a C++ function. `transformer_engine::is_non_tn_fp8_gemm_supported` is a better name than `nvte_is_non_tn_fp8_gemm_supported`.

Reply:
Hm, I see that @ptrendx prefers if we move away from C++ utility functions, and do everything via publicly exposed C APIs. Then the right approach is to define `nvte_is_non_tn_fp8_gemm_supported` within `transformer_engine/common/include/transformer_engine.h`, not in `common.h`. This also means that we can revert `libtransformer_engine.version`, since that exposes all the `nvte_` functions.

Reply:
Do you mean, if we define `nvte_is_non_tn_fp8_gemm_supported` in `transformer_engine/common/include/transformer_engine.h`, then we don't need to explicitly state that it's a C function in `lib*.version`?

Reply:
Also the function is defined in `common.cu` and declared in `common.h`. Is it fine to define it in `transformer_engine.h`?

Reply:
`transformer_engine.h` is reasonable, although it would be even better to have a `utils.h` header within `transformer_engine/common/include/transformer_engine`. We can move things around later though.